fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/TableScanDesc.java

```diff
@@ -56,6 +60,30 @@ public LogicalScanOperator getScanOperator() {
         return scanOperator;
     }
 
+    public boolean isMatch(TableScanDesc other) {
+        boolean matched =  table.equals(other.table);
+        if (!matched) {
+            return false;
+        }
+
+        // for
+        // query: a left join c
+        // mv: a inner join b left join c
+        if (parentJoinType.isInnerJoin()) {
```

Comment:
query: a left join c
mv: a inner join b inner join c
It looks like will match, is it right?

Reply:
c will not match.